src/gifcodec.c

```diff
@@ -303,12 +310,19 @@ gdip_load_gif_image (void *stream, GpImage **image, BOOL from_file)
 	result = NULL;
 	loop_counter = FALSE;
 
+#if GIFLIB_MAJOR < 5
 	if (from_file) {
 		gif = DGifOpen(stream, &gdip_gif_fileinputfunc);
 	} else {
 		gif = DGifOpen (stream, &gdip_gif_inputfunc);
 	}
-	
+#else
+	if (from_file)
+		gif = DGifOpen(stream, &gdip_gif_fileinputfunc, NULL);
+	else
+		gif = DGifOpen(stream, &gdip_gif_inputfunc, NULL);
+#endif
+
```

Comment:
same as what you did on line 49 and 50. But that's just my personal opinion here :)